builtin/providers/aws/resource_aws_security_group.go

```diff
@@ -876,3 +878,68 @@ func protocolForValue(v string) string {
 	log.Printf("[WARN] Unable to determine valid protocol: no matching protocols found")
 	return protocol
 }
+
+func getLambdaENIs(conn *ec2.EC2, d *schema.ResourceData) []ec2.NetworkInterface {
+	filter := &ec2.Filter{
+		Name:   aws.String("group-id"),
+		Values: []*string{aws.String(d.Id())},
+	}
+	params := &ec2.DescribeNetworkInterfacesInput{
+		Filters: []*ec2.Filter{filter},
+	}
+	networkInterfaceResp, err := conn.DescribeNetworkInterfaces(params)
+
+	lambdaENIs := make([]ec2.NetworkInterface, 0)
+	if err != nil {
+		log.Printf("[WARN] Unable to fetch ENIs: %s", err)
+		return lambdaENIs
+	}
+
+	v := networkInterfaceResp.NetworkInterfaces
+	for _, eni := range v {
+		if strings.Contains(*eni.Description, "AWS Lambda VPC ENI") {
```

Comment:
Is there any reason why not put this into the initial filtering on the API level? I think that would possibly make the code slightly cleaner + save some bytes between AWS/user and CPU cycles.

When I'm looking at the real ENI created by Lambda I think we can be even more strict in filtering to be _really_ sure we won't delete any ENI by mistake. It would be a sad coincidence, but after all description is something users can specify (i.e. name it `AWS Lambda VPC ENI`). We want to avoid deleting ENIs that are not managed by the underlying Lambda API if possible.

How about something like this? 😺 

```sh
aws ec2 describe-network-interfaces \
  --filter \
    'Name=group-id,Values=sg-aaaaaaaa' \
    'Name=description,Values=AWS Lambda VPC ENI: *' \
    'Name=requester-id,Values=*:awslambda_*'
```